libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -305,7 +305,12 @@ bool AP_ICEngine::throttle_override(uint8_t &percentage)
     }
 
     if (state == ICE_STARTING || state == ICE_START_DELAY) {
-        percentage = (uint8_t)start_percent.get();
+        int8_t pct = start_percent.get();
+        if (pct < 0) {
```

Comment:
Should check explicitly for `-1` as a flag value here.  That leaves the possibility of using other negative values for other flag values without any risk of changing behaviour for existing users.